### PR TITLE
[Merged by Bors] - refactor: update libs and fix lint issues (VF-3118)

### PR DESCRIPTION
**Pull request by pmvrmc**
**Fixes or implements VF-3118**

### Brief description. What is this change?

Setting up new versions of linter and its plugins and fixing the lint issues

### Implementation details. How do you make this change?

some are fixed through `yarn lint:quiet --fix` others manually

### Related PRs

| project    | branch              | PR          |
| ----------- | ------------------- | ----------- |
| server-data-api | pedro/lint-upgrade/VF-3118 | [link](https://github.com/voiceflow/server-data-api/pull/111) |
| luis-authoring-service | pedro/lint-upgrade/VF-3118 | [link](https://github.com/voiceflow/luis-authoring-service/pull/165) |
| integrations | pedro/lint-upgrade/VF-3118 | [link](https://github.com/voiceflow/integrations/pull/118) |
| google-service | pedro/lint-upgrade/VF-3118 | [link](https://github.com/voiceflow/google-service/pull/244) |
| google-runtime | pedro/lint-upgrade/VF-3118 | [link](https://github.com/voiceflow/google-runtime/pull/206) |
| general-service | pedro/lint-upgrade/VF-3118 | [link](https://github.com/voiceflow/general-service/pull/315) |
| general-runtime | pedro/lint-upgrade/VF-3118 | [link](https://github.com/voiceflow/general-runtime/pull/312) |
| database-cli | pedro/lint-upgrade/VF-3118 | [link](https://github.com/voiceflow/database-cli/pull/190) |
| custom-api | pedro/lint-upgrade/VF-3118 | [link](https://github.com/voiceflow/custom-api/pull/87) |
| creator-api | pedro/lint-upgrade/VF-3118 | [link](https://github.com/voiceflow/creator-api/pull/970) |
| alexa-service | pedro/lint-upgrade/VF-3118 | [link](https://github.com/voiceflow/alexa-service/pull/261) |

### CODE REVIEW

File: lib/services/runtime/programs/resume.ts [hunk 1]
```diff
@@ -14,14 +14,15 @@ export enum ResumeVariables {
   FOLLOW_VOICE = '__voice1__',
 }
 
-export const promptToSSML = (content = '', voice: string | undefined) => {
-  if (!voice || voice === 'Alexa' || !content) {
-    return content;
+export const promptToSSML = (content: string | undefined, voice: string | undefined) => {
```
Comment (z4o4z) [inline lib/services/runtime/programs/resume.ts:17]: can we make voice optional, this should fix lint issues without any code refactoring, isn't it?
```suggestion
export const promptToSSML = (content: string = '', voice? string) => {
```

**Review by z4o4z — APPROVED** [review]

File: lib/services/runtime/programs/resume.ts (hunk 1 above)
Comment (jonahsnider) [reply lib/services/runtime/programs/resume.ts:17]: the same issue will be present, the linter only wants one """optional""" argument (`x: T = y`) which must be the final argument

File: lib/services/runtime/handlers/display/utils.ts [hunk 2]
```diff
@@ -49,5 +49,11 @@ export const getEventToSend = (argument: string) => ({ type: EVENT_SEND_EVENT, a
...
-export const shouldRebuildDisplay = (dataSourceVars: string[] = [], variables: Record<string, any>, lastVariables: Record<string, any> = {}) =>
-  dataSourceVars.some((name) => variables[name] !== lastVariables[name]);
+export const shouldRebuildDisplay = (
+  dataSourceVars: string[] | undefined,
+  variables: Record<string, any>,
+  lastVariables: Record<string, any> = {}
+) => {
+  const dataSourceVariables = dataSourceVars ?? [];
+  return dataSourceVariables.some((name) => variables[name] !== lastVariables[name]);
+};
```
Comment (trs) [inline lib/services/runtime/handlers/display/utils.ts:59]: No need to assign a new variable.

```suggestion
) => (dataSourceVars ?? []).some((name) => variables[name] !== lastVariables[name]);
```

**Review by trs — APPROVED** [review]

**Pull request closed without merging**